addons/base_automation/models/base_automation.py

```diff
@@ -193,8 +196,13 @@ def _process(self, records):
         # execute server actions
         if self.action_server_id:
             for record in records:
-                ctx = {'active_model': record._name, 'active_ids': record.ids, 'active_id': record.id}
-                self.action_server_id.with_context(**ctx).run()
+                # we process the action if we are in a create, or
+                # if we have update_fields_id, we check that we are writing on one of these fields
+                if not self._context.get('old_values') or not self.update_fields_id or (
+                        self._context.get('old_values') and
+                        set(self._context.get('old_values')[record.id]) & set(self.update_fields_id.mapped('name'))):
```

Comment:
I find the condition a bit hard to read. What about this?
```python
old_vals = self._context.get('old_values', {}).get(record.id)
if not old_vals or not self.update_fields_id or any(
    field.name in old_vals for field in self.update_fields_id
):
```
The `any(...)` part translates easily as "some of the trigger fields has been updated."
(Note: this is subjective and reflects personal taste.)

Reply:
It's a much better rewriting. Note that this condition was still false, and I moved it to a function to declutter that part of the code.